lib/algeruby/adt.rb

```diff
@@ -25,6 +25,45 @@ def self.valid_type?(obj)
     obj.is_a?(TypeDescriptor) | (obj.is_a?(Class) && PRIMITIVE_TYPES.include?(obj))
   end
 
+  # This is a top type that matches all values
+  class Anything < TypeDescriptor
```

Comment:
Since validation is based on deserialization and there's a (loose) 1:1 map between descriptor types and the `*Converter` classes in `deser.rb` we probably need those classes too before these new types will be usable for much.